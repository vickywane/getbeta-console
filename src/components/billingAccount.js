import React, { useState } from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'
import { inject, observer } from 'mobx-react'
import { Stack, Select } from '@chakra-ui/core'
import * as Yup from 'yup'
import { Spinner } from 'react-bootstrap'

import useWindowWidth from '../utils/hook_style'
import {
  Text,
  Title,
  center,
  Alert,
  Button,
  Body,
  MdTitle,
  CreateCourseInputField as InputContainer
} from '../styles/style'

const Card = styled.div`
  width: auto;
  margin: 1rem 0;
  span {
    padding: 1rem 1rem;
  }
`

const InputBody = styled.div`
  margin: 1rem 1rem;
  label {
    font-size: 0.85rem;
  }
  input {
    display: flex;
    flex: 1;
    width: ${props => props.width};
    height: 50px;
    border: 1px solid #c0c0c0;
    color: black;
    padding: 0.5rem 1rem;
    border-radius: 3px;
    font-size: 0.9rem;
  }
  select {
    border: 1px solid #c0c0c0;
  }
  ${media.lessThan('large')`
  label {
   font-size : 1rem;
  }
  input {
    display: flex;
    flex: 1;
    width: ${props => props.width};
    height: 40px;
    border: 1px solid ${props => (props.borderColor ? props.borderColor : '#fff')};
    color: blue;
    padding: 0.5rem 1rem;
    border-radius: 3px;
    font-size: 0.9rem;
  }
  ${media.lessThan('medium')`
    margin: 1rem 1rem;
    label {
      font-size : .85rem;
    }
    input {
      font-size : .9rem;
      background: #fbfbfb;
      &:focus {
        box-shadow: 0 0 1.5px 1.5px #0072ce;
      }
    }
  `};
  ${media.lessThan('small')`
  margin: .5rem 1rem;
  label {
    font-size : .8rem;
  }
  input {
    font-size : .8rem;
  }
`};
  `};
`

const FlexInputs = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  place-items: center;
  ${media.lessThan('medium')`
    width : 100%;
    display : flex;
    flex-direction : column;
    align-items : center;
  `}
`

const SmallCard = styled.div`
  ${media.lessThan('medium')`
    width : 95%;
  `};
  ${media.lessThan('small')`
  width : 97%;
`};
`

const billingSchema = Yup.object().shape({
  bankName: Yup.string(),
  cardHolderName: Yup.string(),
  accountNumber: Yup.string()
})

const AlertComponent = ({ show, error }) =>
  show && (
    <Alert style={{ ...center, borderRadius: '5px 5px 0 0', background: error && 'red' }}>
      <Text> Your billing account has been added. </Text>
    </Alert>
  )

const BillingAccountModal = props => {
  const [bank, setBank] = useState('Guaranty Trust Bank') //default
  const [accountName, setAccountName] = useState('')
  const [accountNumber, setAccountNumber] = useState('')
  const [cvv, setCvv] = useState('')
  const [expiry, setExpiry] = useState('')

  const Width = useWindowWidth()
  const {
    subscribeToPlan,
    accountIsUpgraded,
    isLoading,
    errorCreatingBillingAccount,
    createBillingAccount,
    accountIsBillingAccountModald
  } = props.UserStore

  const handleCheckout = () => {
    const isValidAccount = billingSchema.isValid({
      bankName: bank,
      cardHolderName: '',
      accountNumber: ''
    })

    // isValidAccount
    //   .then(() => {
    //   })
    //   .catch(e => console.log(e))

    createBillingAccount(accountName, bank, accountNumber)
  }

  return (
    <div>
      {Width >= 1000 ? (
        <div style={{ ...center }}>
          <div>
            <AlertComponent show={accountIsUpgraded} error={errorCreatingBillingAccount} />
            <Text align="center">
              Billing accounts are only setup once and are used to purchase items on GetBeta{' '}
            </Text>
            <form onSubmit={() => handleCheckout()}>
              <InputBody style={{ color: 'black' }}>
                <label style={{ color: '#fff' }}>BANK NAME</label>
                <Select onChange={e => setBank(e.target.value)}>
                  <option value="Guaranty Trust Bank"> Guaranty Trust Bank </option>
                  <option value="Stanbic Bank"> Stanbic Bank </option>
                  <option value="First Bank Nigeria"> First Bank Nigeria </option>
                </Select>
              </InputBody>

              <InputBody width="100%">
                <label> ACCOUNT NUMBER </label>
                <input
                  value={accountNumber}
                  onChange={e => setAccountNumber(e.target.value)}
                  placeholder="XXXX-XXXX-XXXX-XXXX"
                />
              </InputBody>

              <InputBody width="100%">
                <label> CARDHOLDER NAME</label>

                <input
                  value={accountName}
                  onChange={e => setAccountName(e.target.value)}
                  placeholder="Cardholder Name"
                />
              </InputBody>

              <FlexInputs>
                <InputBody width="100%">
                  <label> EXPIRY DATE </label>
                  <input
                    onChange={e => setExpiry(e.target.value)}
                    value={expiry}
                    placeholder="01-01"
                  />
                </InputBody>

                <InputBody width="100%">
                  <label> CVV </label>
                  <input onChange={e => setCvv(e.target.value)} value={cvv} placeholder="XX-XX" />
                </InputBody>
              </FlexInputs>
            </form>
            <div style={{ display: 'flex', justifyContent: 'center' }}>
              <Button
                onClick={() => handleCheckout()}
                style={{
                  width: '95%'
                }}
              >
                {isLoading ? 'Adding Details' : 'Add Details'}

                {isLoading && (
                  <div style={{ paddingLeft: '.7rem' }}>
                    <Spinner size="sm" animation="border" role="status" />
                  </div>
                )}
              </Button>
            </div>
            <br />
          </div>
        </div>
      ) : (
        <div style={{ ...center }}>
          <SmallCard>
            <AlertComponent show={accountIsUpgraded} error={errorCreatingBillingAccount} />
            <span>
              <form onSubmit={() => handleCheckout()}>
                <InputContainer>
                  <label>Bank Name</label>
                  <Select onChange={e => setBank(e.target.value)}>
                    <option value="Guaranty Trust Bank"> Guaranty Trust Bank </option>
                    <option value="Stanbic Bank"> Stanbic Bank </option>
                    <option value="Gtb"> First Bank Nigeria </option>
                  </Select>
                </InputContainer>

                <InputContainer borderColor="#c0c0c0" width="100%">
                  <label> Account Number </label>
                  <input onChange={e => {}} placeholder="Account Number" />
                </InputContainer>

                <InputContainer borderColor="#c0c0c0" width="100%">
                  <label> Cardholder Name </label>
                  <input placeholder="Cardholder Name" />
                </InputContainer>

                <div style={{ margin: '1rem 0', display: 'flex', justifyContent: 'center' }}>
                  <Button
                    onClick={e => {
                      e.preventDefault()
                      handleCheckout()
                    }}
                    style={{ width: '100%' }}
                  >
                    Add Details
                  </Button>
                </div>
              </form>
            </span>
          </SmallCard>
        </div>
      )}
    </div>
  )
}

export default inject('UserStore')(observer(BillingAccountModal))
