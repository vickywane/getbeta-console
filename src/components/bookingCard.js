import React, { useState } from 'react'
import styled from 'styled-components'
import { navigate } from '@reach/router'
import media from 'styled-media-query'
import { Box, Badge } from '@chakra-ui/core'

import { Text, Title } from '../styles/style'

const UserImage = styled.img`
  height: auto;
  width: 90px;
  cursor: pointer;
  object-fit: cover;
  border-radius: 3px 0 0 3px;
  ${media.lessThan('large')`
  height: auto;
  width: 100px;
  `};
  ${media.lessThan('medium')`
    height: 150px;
    width: 150px;
    border-radius : 50%;
  `};
  ${media.lessThan('small')`
    height: 120px;
    width: 125px;
      border-radius : 5%;
  `};
`

const StyledCard = styled.div`
  width: 30rem;
  height: auto;
  border-radius: 7px;
  background: #fff;
  box-shadow: 0 1px 2px #c0c0c0;
  span {
    display: grid;
    grid-template-columns: 8rem auto;
    grid-gap: 0 0.5rem;
  }
  ${media.lessThan('large')`
      width : 27rem;
  `};
  ${media.lessThan('medium')`
    width: 29rem;
    span {
      padding : 1rem .6rem;
      display: flex;
      flex-direction : column;
      align-items : center;
    }
  `};
  ${media.lessThan('small')`
    width: 20rem;
  `};
`

const BookingCard = props => {
  const { fullname, _id, rating, email, price, setBookingModal, setDetails } = props

  return (
    <StyledCard>
      <span>
        <div style={{ display: 'flex', justifyContent: 'center' }}>
          <UserImage alt="User" src={require('../assets/images/img.jpg')} />
        </div>

        <div style={{ padding: '.5rem 0' }}>
          <div
            style={{
              display: 'flex',
              justifyContent: 'space-between',
              padding: '0 .7rem'
            }}
          >
            <Title
              onClick={() => {
                navigate(`/u/${fullname.trim()}`, {
                  state: { id: _id }
                })
              }}
              style={{ fontWeight: 500, cursor: 'pointer' }}
              color="#0072ce"
              small
              align="center"
            >
              {fullname}
            </Title>
          </div>

          <Text style={{ padding: '0', margin: '0' }} small align="center">
            Lorem ipsum dolor, sit amet consectetur adipisicing elit.{' '}
          </Text>

          <br />

          {/* <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                  <Text small> Slots : 5 </Text>

                  <Text small style={{ color: '#0072ce' }}>
                    ${price} / mins{' '}
                  </Text>
                </div> */}

          <div style={{ display: 'flex', marginTop: '10px', justifyContent: 'space-between' }}>
            <Box d="flex" alignItems="baseline">
              <Badge rounded="full" px="5" fontSize={11} variantColor="teal">
                5 Open Slots
              </Badge>
            </Box>

            <Text
              style={{ paddingRight: '10px' }}
              align="center"
              onClick={() => {
                setBookingModal(true)
                setDetails({
                  fullname: fullname,
                  price: price,
                  email: email,
                  rating: rating
                })
              }}
            >
              Create Booking
            </Text>
          </div>
        </div>
      </span>
    </StyledCard>
  )
}

export default BookingCard
