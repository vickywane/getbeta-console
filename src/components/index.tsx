export { default as Header } from "./header/header"
export { default as Footer } from "./footer"
export { default as Loader } from "./loaders"
export { default as Detail } from "./detail"
export { default as ResolutionError } from "./ResolutionError"
export { default as Panes } from "./panes/pane"
export { default as SettingsPane } from "./panes/settings"
export { default as Switch } from "./switch"
export { default as Checkbox } from './checkbox'
export { default as Tip } from './tip'
