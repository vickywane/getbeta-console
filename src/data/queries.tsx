import { gql } from "apollo-boost"

const USERS: any = gql`
  query users {
    users {
      id
      name
      email
      password
      bucketLink
      createdAt
    }
  }
`

const GET_USER: any = gql`
  query get_user($id: Int!, $name: String!) {
    user(id: $id, name: $name) {
      id
      name
      email
      img_uri
      bucketName
      events {
        id
        name
        eventType
        venue
        alias
        isVirtual
        dateCreated
        summary
        description
        createdBy {
          name
          id
        }
      }
      volunteering {
        role
        id
        duration
        event {
          venue
          id
          name
        }
      }
      attending {
        id
        user {
          name
        }
      }
    }
  }
`

const GET_USER_DETAIL: any = gql`
  query get_user($id: Int!, $name: String!) {
    user(id: $id, name: $name) {
      id
      name
      email
      img_uri
    }
  }
`

const GET_USER_TALKS: any = gql`
  query get_user($id: Int!, $name: String!) {
    user(id: $id, name: $name) {
      id
      talks {
        id
        title
        summary
        createdAt
      }
    }
  }
`

const GET_USER_EVENTS: any = gql`
  query get_user($id: Int!, $name: String!) {
    user(id: $id, name: $name) {
      id
      events {
        id
        name
        venue
      }
    }
  }
`

const GET_EVENT_TALKS: any = gql`
  query GET_EVENT_TALKS($id: Int!, $name: String!) {
    event(id: $id, name: $name) {
      name
      id
      createdAt
      isAcceptingTalks
      speakerConduct
      tracks {
        id
        name
        summary
        duration
        talks {
          id
          title
          speaker {
            id
            name
          }
        }
      }
    }
  }
`

const GET_EVENT: any = gql`
  query GET_EVENT($id: Int!, $name: String!) {
    event(id: $id, name: $name) {
      name
      id
      venue
      description
      summary
      Email
      website
      dateCreated
      bucketName
      alias
      isLocked
      isArchived
      isAcceptingTalks
      isAcceptingVolunteers
      isAcceptingAttendees
      isVirtual
      confirmedEmail
      speakerConduct
      eventType
      mediaLinks
      actions
      EventDate
      sponsors {
        id
        name
        isOrganization
      }
      createdBy {
        name
        email
        id
      }
      teams {
        id
        name
        createdAt
        goal
        tasks {
          id
        }
      }
      tracks {
        id
        name
        summary
        duration
      }
      volunteer {
        role
        id
        approvalStatus
        volunteer_proposal
        dateApplied
        user {
          name
        }
      }
      attendees {
        dateJoined
        user {
          name
          email
        }
      }
      cart_items_category {
        name
        id
      }
      meetupGroups {
        name
        id
        summary
        createdAt
        location
        alias
      }
    }
  }
`

const EVENTS: any = gql`
  query events {
    events {
      id
      name
      summary
      EventDate
      createdBy {
        id
        name
      }
      Email
      venue
      eventType
      description
    }
  }
`

const TEAMS: any = gql`
  query Teams {
    teams {
      id
      name
    }
  }
`

const TEAM: any = gql`
  query team($id: Int!, $name: String!) {
    team(id: $id, name: $name) {
      id
      name
      goal
      createdBy {
        name
      }
      event {
        name
      }
      members {
        name
        id
      }
      tasks {
        name
        category
        createdAt
        createdBy {
          id
          name
        }
      }
    }
  }
`

const TRACKS: any = gql`
  query Track {
    tracks {
      name
      id
    }
  }
`

const GET_TALK: any = gql`
  query talk($id: Int!) {
    talk(id: $id) {
      id
      title
      summary
      description
      tags
      createdAt
      duration
      talkCoverUri
      Archived
    }
  }
`

const TALKS: any = gql`
  query talk {
    talks(Limit: 5) {
      title
      id
      Archived
      description
      summary
      speaker {
        name
      }
    }
  }
`

const GET_VOLUNTEERS: any = gql`
  query get_volunteers {
    volunteers {
      role
      approvalStatus
      duration
      user_description
      user {
        name
      }
    }
  }
`

const GET_CART_ITEM = gql`
  query cartItems($categoryId: Int!) {
    cartItems(categoryId: $categoryId) {
      id
      name
      isFree
      quantity
      price
      description
    }
  }
`

const GET_ALL_EVENT_CART_ITEMS = gql`
  query allCartItems($Limit: Int) {
    allCartItems(Limit: $Limit) {
      id
      name
      isFree
      quantity
      price
      description
    }
  }
`

const GET_CATEGORY = gql`
  query categoryitems($id: Int!) {
    category(id: $id) {
      items {
        id
        name
        isFree
        quantity
        price
        description
      }
    }
  }
`

const GET_EVENT_TALK = gql`
  query GET_EVENT($eventId: Int!, $name: String!) {
    event(id: $eventId, name: $name) {
      talk {
        id
        dateSubmitted
        isAccepted
        draft {
          id
          title
          duration
          summary
          description
          speaker {
            id
            name
            email
          }
        }
      }
      tracks {
        id
        name
        summary
        duration
      }
    }
  }
`

const GET_EVENT_MEETUP_GROUP: any = gql`
  query getMeetupGroup {
    getMeetupGroup(id: 878604100, name: "") {
      name
      createdAt
      summary
      location
      alias
    }
  }
`

export {
  GET_EVENT_TALK,
  GET_USER_TALKS,
  GET_CATEGORY,
  TEAMS,
  GET_ALL_EVENT_CART_ITEMS,
  GET_CART_ITEM,
  GET_USER_DETAIL,
  GET_EVENT_MEETUP_GROUP,
  GET_EVENT_TALKS,
  GET_USER_EVENTS,
  GET_TALK,
  GET_VOLUNTEERS,
  GET_USER,
  USERS,
  TALKS,
  TEAM,
  TRACKS,
  EVENTS,
  GET_EVENT,
}
