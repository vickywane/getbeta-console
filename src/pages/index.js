export { default as Home } from './home/home'
export { default as Bookings } from './bookings/booking'

export { default as Sessions } from './content/sessions/sessions'
export { default as AllSessions } from './content/sessions/allSession'

export { default as Courselist } from './content/courses/courseList'
export { default as CoursePage } from './content/courses/coursePage'

export { default as Settings } from './settings/setting'

export { default as Ticket } from './tickets/ticket'
export { default as Console } from './console'
export { default as CreateSession } from './content/sessions/create-session-card'
export { default as Contents } from './content/content/contentList'
export { default as Content } from './content/content/content'
export { default as EditCourse } from './content/courses/editCourse'

export { default as CreateCourse } from './forms/create-course'

export { default as AccountPlans } from './user/account-plans'
export { default as Analytics } from './user/analytics'
export { default as Upgrade } from './user/upgrade'
export { default as UpdateProfilePage } from './user/updateProfile'

export { default as CreatorsHub } from './user/creators-hub'
export { default as Integrations } from './integrations/integrations'
export { default as CreateVendor } from './vendors/createVendor'
export { default as ContactUser } from './user/contactUser'
export { default as CreateContent } from './forms/create-content'
export { default as Notification } from './user/notifications'
export { default as UserSurvey } from './user/userSurvey'
export {  default as ContentPlayer } from './content/content/player'