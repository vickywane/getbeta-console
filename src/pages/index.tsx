export { default as EventList } from './event/EventList'
export { default as Event } from './event/event'
export { default as Team } from './event/team'
export { default as Console } from './console/console'
// export { default as Signin } from "./auth/login/signin"
export { default as Documentation } from './documentation'
export { default as Media } from './media/media'
export { default as Upload } from './media/upload'
export { default as Editor } from './user/talks/editor'
export { default as Drafts } from './user/talks/drafts'
export { default as Draft } from './user/talks/draft'
export { default as Reminders } from './user/reminders'
export { default as Mobile } from './mobile/mobile'
export { default as Preferences } from './settings/preference.page'
export { default as Schedule } from './event/schedule/schedule'
export { default as SubmitTalk } from './event/submitTalk'

export { default as ThemesMarketplace } from './event/themes/themes.marketplace'

export { default as ProfilePreview } from './user/profilePreview'

//Todo: Naming conflict hxere . UserTalks and EventTalks --->
export { default as ScheduledTalks } from './event/schedule/talks'

export { default as Stream } from './event/stream/stream'
export { default as CreateEvent } from './forms/create-event/create-event'

export { default as Signin } from './forms/auth/main'
export { default as ResetPassword } from './forms/auth/reset-page'

export { default as TaskForm } from './forms/task.form'
export { default as TeamForm } from './forms/team.form'

export { default as MeetupGroup } from './event/meetups/meetupgroup/main'
