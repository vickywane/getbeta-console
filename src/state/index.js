// export {  store as UserStore } from './user.store'
export { default as CourseStore } from './courses.store'
export { default as VendorStore } from './vendor.store'
export { default as ContentStore } from './content.store'
