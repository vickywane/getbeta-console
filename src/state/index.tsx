export { default as ConsoleStore } from "./console.store";
export { default as MediaStore } from "./media.store";
export { default as AuthStore } from "./auth.store";
