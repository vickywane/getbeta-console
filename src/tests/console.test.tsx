import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import { MockedProvider } from "@apollo/react-testing"
import { ApolloProvider } from "@apollo/react-hooks"

import { Console } from "../pages/"
import { ConsoleStore, AuthStore } from "../state/"
import { GET_USER } from "../data/queries"
import client from "../data/config"
import { Loader } from "../components/"

Enzyme.configure({ adapter: new Adapter() })

const mockedData = [
  {
    request: {
      query: GET_USER,
      variables: { id: 5577006791947779410 },
    },
    result: {
      data: {
        ID: 5577006791947779410,
        Name: "John Doe",
        Email: "Johndoe@gmail.com",
        BucketLink: "https://google.cloud.com/storage",
      },
    },
  },
]

describe("Tests the User Console", () => {
  const Component = shallow(
    <MockedProvider mocks={mockedData} addTypename={false}>
      <ApolloProvider client={client}>
        <Console AuthStore={AuthStore} ConsoleStore={ConsoleStore} />
      </ApolloProvider>
    </MockedProvider>
  )

  it("Check that Apollo works correctly", () => {
    const tree = Component
    console.log(tree.children)

    //Todo : check if apollo is loading && if loader component is present

    // expect(tree.children).toContain(<Loader/>)
  })

  it("checks if main items exits", () => {
    const ConsoleWrapper = Component

    expect(ConsoleWrapper.find("h2"))
  })
})
