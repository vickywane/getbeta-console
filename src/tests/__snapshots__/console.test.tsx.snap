// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the User Console checks if main items exits 1`] = `ShallowWrapper {}`;
